deltachat-jsonrpc/src/api/types/chat_list.rs

```diff
@@ -70,10 +70,8 @@ pub(crate) async fn get_chat_list_item_by_id(
         });
     }
 
-    let last_msgid = get_last_message_for_chat(ctx, chat_id).await?;
-
     let chat = Chat::load_from_db(ctx, chat_id).await.context("chat")?;
-    let summary = Chatlist::get_summary2(ctx, chat_id, last_msgid, Some(&chat))
+    let summary = Chatlist::get_summary2(ctx, chat_id, Some(&chat))
```

Comment:
So, the main goal of this PR is to make less queries from the db here and avoid races with disappearing messages (at least when `summary.id` is not used). Before there were 3 calls: `get_last_message_for_chat()`, `Chatlist::get_summary2()` and `Message::load_from_db()`, now there's only `Chatlist::get_summary2()`

Reply:
to me, it looks as if the number of database queries are the same, not less. and the queries are more complex

also note, that things will still be slower:

before: `dc_chat_get_summary(chat_id, lastmsg_id)` calls `Chatlist::get_summary2(chat_id, lastmsg_id)` calls `Message::load_from_db(lastmsg_id)` (fast, query affects one row only)

after: `dc_chat_get_summary(chat_id, lastmsg_id)` calls `Chatlist::get_summary2(chat_id)` calls `Message::load_from_db_last_for_chat(chat_id)` (slow, query affects all messages in the chat)

Reply:
First of all, if you already have `dc_chatlist_t` object, you can use `dc_chatlist_get_summary(chatlist, index, chat)` as before, which uses cached message ids. I guees this is the case for Android/iOS, please correct me if i'm wrong.

As for this particular code (which in fact doesn't use chatlist objects!), please note that i removed a call to `get_last_message_for_chat()` which did
```
            "SELECT id                                                                                                                                                                                               
                FROM msgs                                                                                                                                                                                            
                WHERE chat_id=?2                                                                                                                                                                                     
                AND (hidden=0 OR state=?1)                                                                                                                                                                           
                ORDER BY timestamp DESC, id DESC LIMIT 1"
```
(FWIW we have `CREATE INDEX msgs_index3 ON msgs (timestamp);`, so i guess it's not slow). But as now `Chatlist::get_summary2(chat_id)` calls `Message::load_from_db_last_for_chat(chat_id)`, which does the same:
```
                    "WHERE m.chat_id=? ",
                    "AND (m.hidden=0 OR m.state=?) ",
                    "ORDER BY m.timestamp DESC, m.id DESC LIMIT 1",
```
, in fact i just merged two queries into one, so i don't think it is slower than before, but the race involving possibly expired message id goes away.

Additionally, if you look below, i added all necessary info to `struct Summary`, so one more call to `Message::load_from_db()` went away.